packages/javascript-wrapper/lib/utils/setup.js

```diff
@@ -132,7 +132,7 @@ setup.createOption('requestHeaders', 'object', {});
 setup.createOption('suppressAccessTracking', 'boolean', false);
 
 var btoa = function (string) {
-    return Buffer(string).toString('base64');
+    return Buffer.from(string, 'binary').toString('base64');
```

Comment:
According to [the documentation](https://nodejs.github.io/nodejs.dev/en/api/v18/buffer/#buffers-and-character-encodings) `binary` is legacy alias for `latin1`. And `utf8` is default value, so it is better to pass `utf8` as second parameter or omit it.

Reply:
We cannot use `Buffer.from(string, 'utf8')` here, because in combination with `unescape(encodeURIComponent(...))` it will produce different result.

Although `Buffer(string, 'binary').toString('base64')` works like `window.btoa(...)`, i agree not to use it due to its legacy 👍🏻.

So in second iteration i suggest some refactoring of `setup.encodeForYt` method, where:
 - `window.btoa(unescape(encodeURIComponent(str)))` will be used for browser environment (as well known [browser hack](https://ecmanaut.blogspot.com/2006/07/encoding-decoding-utf8-in-javascript.html) to encode utf-8)
 - and `Buffer.from(string).toString('base64')` for Node js utf-8 encoding